tests/core/test_lightning_optimizer.py

```diff
@@ -0,0 +1,192 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+from unittest.mock import patch
+
+import pytest
+import torch
+import torch.nn as nn
+from torch.optim import Adam, Optimizer
+
+from pytorch_lightning import LightningModule, Trainer
+from pytorch_lightning.core.optimizer import LightningOptimizer
+from tests.base.boring_model import BoringModel, RandomDictDataset, RandomDictStringDataset
+
+
+def test_lightning_optimizer(tmpdir):
+    """
+    Test that optimizer are correctly wrapped by our LightningOptimizer
+    """
+    class TestModel(BoringModel):
+        def configure_optimizers(self):
+            optimizer = torch.optim.SGD(self.layer.parameters(), lr=0.1)
+            # optimizer = LightningOptimizer(self.trainer, optimizer)
+            lr_scheduler = torch.optim.lr_scheduler.StepLR(optimizer, step_size=1)
+            return [optimizer], [lr_scheduler]
+    model = TestModel()
+    trainer = Trainer(
+        default_root_dir=os.getcwd(),
+        limit_train_batches=1,
+        limit_val_batches=1,
+        max_epochs=1,
+        weights_summary=None,
+        enable_pl_optimizer=True)
+    trainer.fit(model)
+
+    groups = "{'dampening': 0, 'initial_lr': 0.1, 'lr': 0.01, 'momentum': 0, 'nesterov': False, 'weight_decay': 0}"
+    expected = f"LightningSGD(groups=[{groups}])"
+    assert trainer.optimizers[0].__repr__() == expected
+
+
+def test_lightning_optimizer_from_user(tmpdir):
+    """
+    Test that the user can use our LightningOptimizer. Not recommended.
+    """
+
+    class TestModel(BoringModel):
+        def configure_optimizers(self):
+            optimizer = torch.optim.Adam(self.layer.parameters(), lr=0.1)
+            optimizer = LightningOptimizer(optimizer)
+            lr_scheduler = torch.optim.lr_scheduler.StepLR(optimizer, step_size=1)
+            return [optimizer], [lr_scheduler]
+    model = TestModel()
+    trainer = Trainer(
+        default_root_dir=os.getcwd(),
+        limit_train_batches=1,
+        limit_val_batches=1,
+        max_epochs=1,
+        weights_summary=None,
+        enable_pl_optimizer=True)
+    trainer.fit(model)
+
+    groups = "{'amsgrad': False, 'betas': (0.9, 0.999), 'eps': 1e-08, 'initial_lr': 0.1, 'lr': 0.01, 'weight_decay': 0}"
```

Comment:
the same, let's define defaults explicitly

Reply:
same here